File: 은지/LV2/카펫.js
function solution(brown, yellow) {
  var s = brown + yellow
  for (let x=s; x>=1; x--) {
    if (s % x === 0 && (x-2) * (parseInt(s/x)-2) === yellow) {
      var answer = [x, parseInt(s/x)]
      break
    }
  }
  return answer;
}

File: 은지/LV2/카펫.py
def solution(brown, yellow):
    s = brown + yellow
    for x in range(s, 0, -1):
        if s % x == 0:
            w = x
            h = s // x
            if (w-2) * (h-2) == yellow:
                answer = [w, h]
                break
    return answer

b = 10
y = 2
print(solution(b, y))